WordPress/src/main/java/org/wordpress/android/ui/sitecreation/verticals/NewSiteCreationVerticalsViewModel.kt

```diff
@@ -259,15 +269,21 @@ class NewSiteCreationVerticalsViewModel @Inject constructor(
                             R.string.new_site_creation_verticals_custom_subtitle,
                             showDivider = index != lastItemIndex
                     )
-                    itemUiState.onItemTapped = { _verticalSelected.value = itemUiState.id }
+                    itemUiState.onItemTapped = {
```

Comment:
How would you feel about doing a little bit of cleanup inside the `else` statement to make the code a bit more readable? I had the following in mind:

```
val lastItemIndex = data.size - 1
data.forEachIndexed { index, model ->
    val onItemTapped = {
        tracker.trackVerticalSelected(model.name, model.verticalId, model.isUserInputVertical)
        _verticalSelected.value = model.verticalId
    }
    val itemUiState = if (model.isUserInputVertical) {
        VerticalsCustomModelUiState(
                model.verticalId,
                model.name,
                R.string.new_site_creation_verticals_custom_subtitle,
                showDivider = index != lastItemIndex
        )
    } else {
        VerticalsModelUiState(
                model.verticalId,
                model.name,
                showDivider = index != lastItemIndex
        )
    }
    itemUiState.onItemTapped = onItemTapped
    items.add(itemUiState)
}
```

I think we could clean it up a little further, but it's probably diminishing returns after this.

Reply:
Definitely a good call!

Fixed in [#a8187a9](https://github.com/wordpress-mobile/WordPress-Android/pull/8998/commits/a8187a93e08226846a0711a53ba23bfd29281f32)